src/main/daml/Daml/Finance/Account/Account.daml

```diff
@@ -89,9 +85,15 @@ template Account
 
     interface instance Disclosure.I for Account where
       view = Disclosure.View with disclosureControllers = S.fromList [custodian, owner]; observers
-      setObservers = setObserversImpl this $ Some (Account.disclosureUpdateReference account)
-      addObservers = addObserversImpl this $ Some (Account.disclosureUpdateReference account)
-      removeObservers = removeObserversImpl this $ Some (Account.disclosureUpdateReference account)
+      setObservers = setObserversImpl this $ Some (Account.disclosureUpdateReference (accountKey this))
+      addObservers = addObserversImpl this $ Some (Account.disclosureUpdateReference (accountKey this))
+      removeObservers = removeObserversImpl this $ Some (Account.disclosureUpdateReference (accountKey this))
+
+accountKey : Account -> AccountKey
```

Comment:
I would `--| HIDE` the new utility functions so that they don't pop up in the docs

Reply:
thanks, I addressed this in a new commit